### ISSUE TITLE
[Bug]: [JSON] The result is empty when searching/querying with expression that float data compared with integar

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#24226]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20230518-af271498
- Deployment mode(standalone or cluster):both
- MQ type(rocksmq, pulsar or kafka):    all
- SDK version(e.g. pymilvus v2.0.0rc2):pymilvus 2.4.0.dev35
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

The result is empty when searching/querying with expression that float data compared with integar
(1) int-int: good
res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"number\"] <= 10", output_fields=["json_field"])
(2) int - float: good
res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"number\"] <= 10.0", output_fields=["json_field"])
(3) float - float: good
res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"float\"] <= 10.0", output_fields=["json_field"])
(4) float- int: bad
res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"float\"] <= 10", output_fields=["json_field"])

```
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"float\"] <= 10.0", output_fields=["json_field"])
>>> res[0].ids
[0, 2218, 1069, 133, 1883, 673, 1155, 597, 2189, 1544]
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"float\"] <= 10", output_fields=["json_field"])
>>> res[0].ids
[]
```

### Expected Behavior

Search successfully

### Steps To Reproduce

```python
dim = 128
int64_field = FieldSchema(name="int64", dtype=DataType.INT64, is_primary=True)
float_field = FieldSchema(name="float", dtype=DataType.FLOAT)
bool_field = FieldSchema(name="bool", dtype=DataType.BOOL)
string_field = FieldSchema(name="string", dtype=DataType.VARCHAR, max_length=65535)
json_field = FieldSchema(name="json_field", dtype=DataType.JSON)
json_field_1 = FieldSchema(name="json_field_1", dtype=DataType.JSON)
float_vector = FieldSchema(name="float_vector", dtype=DataType.FLOAT_VECTOR, dim=dim)
schema = CollectionSchema(fields=[int64_field, float_field, bool_field, json_field, json_field_1, float_vector])
collection = Collection("test_search_collection_multiple_json_0", schema=schema)
import random
nb=3000
vectors = [[random.random() for _ in range(dim)] for _ in range(nb)]
import numpy as np
res = collection.insert([[i for i in range(nb)], [np.float32(i) for i in range(nb)], [np.bool_(i) for i in range(nb)], [{"number": i, "float": 1.111111111111111111111111111, "int64":  1000000000000000180980893038800089, "string": "milvus", "bool": True, "list": [j for j in range(0, i)], "json": {"number": i, "string": "milvus", "bool": True}} for i in range(0, nb)], [{"number": i, "string": "milvus", "bool": True, "list": [j for j in range(0, i)], "json": {"number": i, "string": "milvus", "bool": True}} for i in range(0, nb)], vectors])
index_param = {"index_type": "IVF_SQ8", "metric_type": "L2", "params": {"nlist": 8192}}
collection.create_index("float_vector", index_param, index_name="index_name_1")
collection.load()
default_search_params = {"metric_type": "L2", "params": {"nprobe": 10}}
limit = 10
nq =1
res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"number\"] <= 10", output_fields=["json_field"])
res[0][0]
res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"number\"] <= 10.0", output_fields=["json_field"])
res[0][0]
res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"float\"] <= 10.0", output_fields=["json_field"])
res[0][0]
res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"float\"] <= 10", output_fields=["json_field"])
res[0][0]
```

**Comment by yah01** [comment]
We need to allow type convertion with the json field

**Comment by yah01** [comment]
/assign @binbinlv 
Fixed with #24229

**Comment by binbinlv** [comment]
Verified and fixed in master branch:
milvus: master-20230519-3a66e1de
pymilvus: 2.4.0.dev39

```
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"float\"] <= 10", output_fields=["json_field"])
>>> res[0][0]
id: 0, distance: 0.0, entity: {'json_field': b'{"number":0,"float":1.1111111111111112,"int64":1000000000000000180980893038800089,"string":"milvus","bool":true,"list":[],"json":{"number":0,"string":"milvus","bool":true}}'}
>>>
>>>
>>> res[0].ids
[0, 2515, 1753, 2751, 14, 2036, 2830, 2634, 1074, 1130]
```

**Comment by binbinlv** [comment]
Verified and fixed in 2.2.* branch:
milvus: [2.2.0-20230520-1eb804fc](https://hub.docker.com/layers/milvusdb/milvus/2.2.0-20230520-1eb804fc/images/sha256-f1fce0b4ba76324119d89cebe860886197da341061e6688f0ef9da34719ef485?context=explore)
pymilvus: 2.2.9.dev12

```
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"float\"] <= 10.0", output_fields=["json_field"])
>>>
>>> res[0][0]
id: 0, distance: 0.0, entity: {'json_field': b'{"number":0,"float":1.1111111111111112,"int64":1000000000000000180980893038800089,"string":"milvus","bool":true,"list":[],"json":{"number":0,"string":"milvus","bool":true}}'}
>>>
>>> res[0].ids
[0, 2637, 1250, 2307, 2250, 2178, 1277, 592, 318, 2518]
>>>
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"float\"] <= 10", output_fields=["json_field"])
>>>
>>> res[0].ids
[0, 2637, 1250, 2307, 2250, 2178, 1277, 592, 318, 2518]
>>>
>>> res[0][0]
id: 0, distance: 0.0, entity: {'json_field': b'{"number":0,"float":1.1111111111111112,"int64":1000000000000000180980893038800089,"string":"milvus","bool":true,"list":[],"json":{"number":0,"string":"milvus","bool":true}}'}
>>>
```
